streams/src/test/java/org/apache/kafka/streams/kstream/JoinWindowsTest.java

```diff
@@ -85,14 +86,55 @@ public void timeDifferenceMustNotBeNegative() {
         JoinWindows.of(-1);
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void afterBelowLower() {
-        JoinWindows.of(anySize).after(-anySize - 1);
+        final JoinWindows windowSpec = JoinWindows.of(anySize);
+        try {
+            windowSpec.after(-anySize - 1);
+            fail("window end time should not be before window start time");
+        } catch (final IllegalArgumentException e) {
+            // expected
+        }
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void beforeOverUpper() {
```

Comment:
i'd rename this to something more descriptive, too. Though i just realised this and the last haven't changed in this PR!